test/cpp/end2end/xds_end2end_test.cc

```diff
@@ -1028,6 +1050,176 @@ TEST_F(SingleBalancerTest, AllServersUnreachableFailFast) {
+                               << " message=" << status.error_message();
+      EXPECT_EQ(response.message(), kRequestMessage_);
+    }
+  }
+  gpr_log(GPR_INFO, "========= DONE WITH FIRST BATCH ==========");
+  // The drop rate should be roughly equal to the expectation.
+  double seen_drop_rate = static_cast<double>(num_drops) / kNumRpcs;
+  const double error_tolerance = 0.2;
+  EXPECT_THAT(seen_drop_rate,
+              ::testing::AllOf(
+                  ::testing::Ge(drop_rate_for_lb * (1 - error_tolerance)),
+                  ::testing::Le(drop_rate_for_lb * (1 + error_tolerance))));
+  // Wait until the drop rate increases to the middle of the two configs, which
+  // implies that the update has been in effect.
+  const double drop_rate_threshold =
+      (drop_rate_for_lb + drop_rate_for_lb_and_throttle) / 2;
+  size_t num_rpcs = kNumRpcs;
+  while (seen_drop_rate < drop_rate_threshold) {
+    EchoResponse response;
+    const Status status = SendRpc(&response);
+    ++num_rpcs;
+    if (!status.ok() &&
+        status.error_message() == "Call dropped by load balancing policy") {
+      ++num_drops;
+    } else {
+      EXPECT_TRUE(status.ok()) << "code=" << status.error_code()
+                               << " message=" << status.error_message();
+      EXPECT_EQ(response.message(), kRequestMessage_);
+    }
+    seen_drop_rate = static_cast<double>(num_drops) / num_rpcs;
+  }
+  // Send kNumRpcs RPCs and count the drops.
+  num_drops = 0;
+  gpr_log(GPR_INFO, "========= BEFORE SECOND BATCH ==========");
+  for (size_t i = 0; i < kNumRpcs; ++i) {
+    EchoResponse response;
+    const Status status = SendRpc(&response);
+    if (!status.ok() &&
+        status.error_message() == "Call dropped by load balancing policy") {
+      ++num_drops;
+    } else {
+      EXPECT_TRUE(status.ok()) << "code=" << status.error_code()
+                               << " message=" << status.error_message();
+      EXPECT_EQ(response.message(), kRequestMessage_);
+    }
+  }
+  gpr_log(GPR_INFO, "========= DONE WITH SECOND BATCH ==========");
+  // The new drop rate should be roughly equal to the expectation.
+  seen_drop_rate = static_cast<double>(num_drops) / kNumRpcs;
+  EXPECT_THAT(seen_drop_rate,
+              ::testing::AllOf(::testing::Ge(drop_rate_for_lb_and_throttle *
+                                             (1 - error_tolerance)),
+                               ::testing::Le(drop_rate_for_lb_and_throttle *
+                                             (1 + error_tolerance))));
+  // The EDS service got a single request,
+  EXPECT_EQ(1U, balancers_[0]->eds_service()->request_count());
+  // and sent two responses
+  EXPECT_EQ(2U, balancers_[0]->eds_service()->response_count());
+}
+
+TEST_F(SingleBalancerTest, DropAll) {
+  SetNextResolution({}, kDefaultServiceConfig_.c_str());
+  SetNextResolutionForLbChannelAllBalancers();
+  const size_t kNumRpcs = 1000;
+  const uint32_t drop_per_million_for_lb = 100000;
```

Comment:
Let's have this set up such that neither category individually drops 100% but the two categories between them drop 100% (e.g., "lb" drops 20% and "throttle" drops 80%).  That way, we know we'll handle that case right.